src/index.js

```diff
@@ -1,42 +1,62 @@
-import wasm from "./wasm";
+const {wasmInit}  = require("./wasm");
+
+let wasm;
+
+/**
+ * Use this method to initialize wasm build.
+ * This method needs to be called only once but
+ * before using any other method.
+ */
+export async function initSha256() {
+  if(!wasm) {
+    wasm = await wasmInit();
+  }
+}
 
 /**
  * Wraps the AssemblyScript build in a JS function.
  * This allows users to not have to make AS a dependency in their project.
  * @param {Uint8Array} message Message to hash
  */
-export default function sha256(message) {
-  // @ts-ignore
-  const buf = Buffer.from(message);
-  const arr = wasm.__retain(wasm.__allocArray(wasm.UINT8ARRAY_ID, buf));
```

Comment:
I don't see any significant changes in wasm file which could potentially corrupt result. So need look at host changes. Did you try revert `Buffer.from(message)` ?

Reply:
Yeah, it's not related to Buffer, that line here was useless.

We were on pretty old version (targetting commit #5b510571f6ebfc5530d85412005059e4cf961a97). The whole code works if I  just switch assemblyscript to old version but then it fails on browsers because old `instantiate` method is using `new WebAssembly.Instance` instead of `await WebAssembly.instantiate(module)`

Reply:
In browser you should use `loader.instantiateStreaming` with latest AS: https://docs.assemblyscript.org/basics/loader#example

Reply:
that method is broken when using in react or angular (webpack) because it breaks on invalid MIME type and it makes usage of library weird as you have to copy wasm blob from library and put somewhere on path.

That method is useful if you are using wasm code in end product

Reply:
If you have advice what url to put as param to `fetch` and how to fix MIME type so it works in webpack bundles would be appreciated :sweat_smile:

Reply:
Hmm, I thought  webpack-dev-server fixed issue with MIME wasm type for a long type age

Reply:
Ok, I made 2 suggestions. This should fix tests

Reply:
Checked it, it seems that change was introduced in v3.3.0 while create-react-app and angular cli still uses <= 3.2.1.

Still the url path is problem, I could expose it as env variable but it's a bit ugly solution and not worth it for wasm blob thats <=7kb like this